### PR TITLE
Add nms_rotated ort op

### PR DESCRIPTION
**Pull request by zytx121**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add nms_rotated ort op.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: csrc/backend_ops/onnxruntime/nms_rotated/nms_rotated.cpp [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+  const float iou_threshold = iou_threshold_;
+
+  const OrtValue *boxes = ort_.KernelContext_GetInput(context, 0);
+  const float *boxes_data =
+      reinterpret_cast<const float *>(ort_.GetTensorData<float>(boxes));
+  const OrtValue *scores = ort_.KernelContext_GetInput(context, 1);
+  const float *scores_data =
+      reinterpret_cast<const float *>(ort_.GetTensorData<float>(scores));
+
+  OrtTensorDimensions boxes_dim(ort_, boxes); // 得到输入的维度
```
Comment (grimoire) [inline csrc/backend_ops/onnxruntime/nms_rotated/nms_rotated.cpp:305]: Please avoid using chinese comment.

File: csrc/backend_ops/onnxruntime/nms_rotated/nms_rotated.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  ONNXTensorElementDataType GetOutputType(size_t) const {
+    return ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64;
+  }
+
+  // force cpu
+  const char* GetExecutionProviderType() const { return "CPUExecutionProvider"; }
+};
+}  // namespace mmdeploy
+
+#endif  // ONNXRUNTIME_NMS_ROTATED_H
```
Comment (AllentDan) [inline csrc/backend_ops/onnxruntime/nms_rotated/nms_rotated.h:48]: need an empty line

File: csrc/backend_ops/onnxruntime/nms_rotated/nms_rotated.cpp [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+#include <cassert>
+
+#include "ort_utils.h"
+
+namespace mmdeploy {
+
+struct RotatedBox {
+  float x_ctr, y_ctr, w, h, a;
+};
+struct Point {
```
Comment (lzhangzz) [inline csrc/backend_ops/onnxruntime/nms_rotated/nms_rotated.cpp:21]: Names like `Point`, `dot_2d`, `cross_2d` are very common and may lead to name collsion in the future. Put op specific internal implementations inside a anonymous namespace to force internal linkage for them, like
```C++
namespace {

struct RotatedBox {...};
struct Point {...};

// ...

}

// and use them normally in the same cpp file
```

**Review by AllentDan — COMMENTED** [review]
May update the supported ops in [onnxruntime doc](https://github.com/open-mmlab/mmdeploy/blob/master/docs/en/backends/onnxruntime.md)

**Review by grimoire — DISMISSED** [review]
LGTM

**Comment by irexyc** [comment]
LGTM

**Review by irexyc — DISMISSED** [review]

**Pull request merged**